Sources/Tracing/Docs.docc/Guides/TraceYourApplication.md

```diff
@@ -0,0 +1,443 @@
+# Trace Your Application
+
+## Overview
+
+This guide is aimed at **application developers** who have some server-side system and want to make use of distributed tracing
+in order to improve their understanding and facilitate performance tuning and debugging their services in production or development.
+
+Distributed tracing offers a way to gain additional insight into how your application is performing in production, without having to reconstruct the "big picture" from manually piecing together log lines and figuring out what happened
+after what else and _why_. Distributed traces, as the name implies, also span multiple nodes in a microservice architecture
+or clustered system, and provide a profiler-like experience to debugging the handling of a "request" or otherwise defined span.
+
+### Setting up
+
+The first step to get metadata propagation and tracing working in your application is picking an instrumentation or tracer 
+implementation. A complete [list of swift-distributed-tracing implementations](http://github.com/apple/swift-distributed-tracing) 
+is available in this project's README. Select an implementation you'd like to use and follow its bootstrap steps.
+
+> Note: Since instrumenting an **application** in practice will always need to pull in an existing tracer implementation,
+> in this guide we'll use the community maintained [`opentelemetry-swift`](https://github.com/slashmo/opentelemetry-swift) 
+> tracer, as an example of how you'd start using tracing in your real applications.
+> 
+> If you'd rather implement your own tracer, refer to <doc:ImplementATracer>.
+
+Once you have selected an implementation, add it as a dependency to your `Package.swift` file. 
+
+```swift
+// Depend on the instrumentation library, e.g. opentelemetry-swift:
+.package(url: "https://github.com/slashmo/opentelemetry-swift.git", from: "<latest-version>"),
+
+// This will automatically include a dependency on the swift-distributed-tracing API:
+// .package(url: "https://github.com/apple/swift-distributed-tracing.git", from: 1.0.0),
+```
+
+Next, add the dependency to your application target:
+
+```swift
+.target(
+    name: "MyApplication",
+    dependencies: [
+      .product(name: "OpenTelemetry", package: "opentelemetry-swift"),
```

Comment:
Maybe should include `OtlpGRPCSpanExporting` from very start?

Reply:
For what it's worth this might better be "SomeTracingSystem" and not the otel not to confuse people. We just happen to use otel in the examples below, but this just explains adding a dependency 🤔  I'll think some more how to do this well.